benchmark/src/main/java/com.google.cloud.bigquery/ConnImplBenchmark.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.bigquery;
+
+import java.io.IOException;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+@Fork(value = 1)
+@BenchmarkMode(Mode.AverageTime)
+@Warmup(iterations = 0)
+@Measurement(iterations = 3)
+@State(Scope.Benchmark)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+public class ConnImplBenchmark {
+  @Param({"500000", "1000000", "10000000", "100000000"}) // 500K, 1M, 10M, and 100M
+  public int rowLimit;
+
+  private ConnectionSettings connectionSettingsReadAPIEnabled, connectionSettingsReadAPIDisabled;
```

Comment:
Perhaps worth filing a similar issue for Java, if populating `connectionProperties` isn't included in this version. I think it'd be the best user experience if a session was automatically created and passed along by the driver.

Reply:
We have both `session` and `connectionProperties` in `ConnectionSettings` already:

connectionProperties:
```
    /**
     * Sets a connection-level property to customize query behavior.
     *
     * @param connectionProperties connectionProperties or {@code null} for none
     */
    public abstract Builder setConnectionProperties(List<ConnectionProperty> connectionProperties);
```

setCreateSession:
```
    /**
     * Sets whether to create a new session. If {@code true} a random session id will be generated
     * by BigQuery. If false, runs query with an existing session_id passed in ConnectionProperty,
     * otherwise runs query in non-session mode."
     */
    public abstract Builder setCreateSession(Boolean createSession);
```